server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -102,60 +102,41 @@ async def create_model_endpoint(
         if not model_endpoint.metadata.uid:
             model_endpoint.metadata.uid = uuid.uuid4().hex
 
-        if not model_endpoint.spec.function_uid:
-            # get function_uid from db
-            try:
-                logger.info("Getting function uid from db")
-                current_function = await run_in_threadpool(
-                    framework.utils.singletons.db.get_db().get_function,
-                    db_session,
-                    name=model_endpoint.spec.function_name,
-                    tag=model_endpoint.spec.function_tag,
-                    project=model_endpoint.metadata.project,
-                )
-                model_endpoint.spec.function_uid = current_function.get(
-                    "metadata", {}
-                ).get("uid")
-            except mlrun.errors.MLRunNotFoundError:
-                logger.info("The model endpoint is created on a non-existing function")
-        model_obj = None
+        model_obj, model_uri = None, None
+        model_path = model_endpoint.spec.model_path
         if model_path and mlrun.datastore.is_store_uri(model_path):
             _, model_uri = mlrun.datastore.parse_store_uri(model_path)
             project, key, iteration, tag, tree, uid = parse_artifact_uri(
                 model_uri, model_endpoint.metadata.project
             )
-        else:
-            project, key, iteration, tag, tree, uid = (
-                model_endpoint.metadata.project,
-                model_endpoint.spec.model_db_key,
-                None,
-                model_endpoint.spec.model_tag,
-                None,
-                model_endpoint.spec.model_uid,
-            )
-        try:
-            logger.info("Getting model object from db")
-            model_obj = mlrun.artifacts.dict_to_artifact(
-                services.api.crud.Artifacts().get_artifact(
+            try:
+                logger.info("Getting model object from db")
+                # Retrieve the model object from the database to extract its ID.
+                # The ID is later used to link the model endpoint to the model object.
+                # Fetching it here prevents retrieving the model object twice.
+                db_artifact = framework.utils.singletons.db.get_db().read_artifact(
                     db_session,
-                    key=key,
-                    tag=tag,
-                    iter=iteration,
-                    project=project,
-                    producer_id=tree,
-                    object_uid=uid,
+                    key,
+                    tag,
+                    iteration,
+                    project,
+                    tree,
+                    uid,
+                    as_record=True,
```

Comment:
I don't see `as_record` param in `read_artifact`. Are you sure it's valid?

Reply:
Yep it is.. you can see under `sql/db.py`.

Reply:
oh yeah but it's not exist in the abstract class within `db/base.py` but I guess you can leave it for now because it doesn't related to your PR.